Documentation/gettingstarted/local-redirect-policy.rst

```diff
@@ -301,5 +305,140 @@ configuring the CiliumLocalRedirectPolicy.
+  traffic from the kiam agent to the AWS metadata server without any redirection,
+  we need the socket lookup functionality in the datapath. This functionality
+  requires v5.1.16, v5.2.0 or more recent Linux kernel. Make sure the kernel
+  version installed on EKS cluster nodes satisfies these requirements.
+
+- Deploy kiam and local redirect policy.
+
+  .. tabs::
+
+    .. group-tab:: Quick Start
+
+            Deploy kiam using the example yaml file.
+
+            .. parsed-literal::
+
+                $ kubectl apply -f \ |SCM_WEB|\/examples/kubernetes-local-redirect/kiam.yaml -n kube-system
+
+            Deploy the local redirect policy to redirect pod traffic to the
+            deployed kiam agent.
+
+            .. parsed-literal::
+
+                $ kubectl apply -f \ |SCM_WEB|\/examples/kubernetes-local-redirect/kiam-lrp.yaml
+
+    .. group-tab:: Manual Configuration
+
+        - Add a metadata label for the kiam agent daemonset.
+
+        - Remove the "--iptables" argument passed to the kiam-agent Daemonset.
+
+        - Make sure the ``kiam-agent Daemonset`` is run in the ``hostNetwork`` mode.
+
+      .. note::
+
+          - The ``localEndpointSelector`` label specified in the local redirect
+            policy needs to match with the metadata label added for the kiam agent
+            daemonset above.
+          - The ``addressMatcher`` ip address in the local redirect policy is set to
+            the ip address of the AWS metadata server and the ``toPorts`` port
+            to the default HTTP server port. The ``toPorts`` field under
+            ``redirectBackend`` configuration in the policy is set to the port that
+            the kiam agent listens on.
+          - The local redirect policy namespace is set to the namespace
+            in which kiam-agent Daemonset is deployed.
+
+
+- Once all the kiam agent pods are in ``Running`` state, the metadata requests
+  from application pods should get redirected to the node-local kiam agent pods.
+  You can verify this by running ``cilium monitor`` with ``related-to`` set to
+  the endpoint id of one of the application pods.
+
+Miscellaneous
+=============
+When a local redirect policy is enabled, cilium BPF datapath translates frontend
+ip address from the policy to the ip address of the node-local backend pod selected
+by the policy. However, in certain cases, it might be desirable to skip
+redirecting traffic in order to avoid packets getting looped back to the backend
+pod. This can happen when the backend pod tries to connect to the frontend ip
+address, and the traffic is translated to the ip address of the backend pod.
+In such cases, we use the Linux kernel `socket lookup functionality <https://lwn.net/Articles/767134/>`_
+to identify sockets listening on the backend pod's ip/port/protocol tuple in the
+same network namespace where the backend pod traffic originates from, and
+skip service translation.
```

Comment:
@borkmann were you advocating for these details in the GSG?

IMO this seems a bit too deep into implementation details, and I worry that as we add more and more specific details into the getting started guides that we end up providing the impression that Cilium is too complicated.

This is great, useful information if you're a developer with a problem to solve for which this happens to be the solution. If you're a user, I don't think you can do anything with this information. 

```
If we phrased this differently under the limitations section, maybe something like "On kernels earlier than 5.1, connections which are proxied via a local pod via LRP may be broken because the kernel cannot distinguish the inbound connection to the pod vs. the outbound connection from the pod. Upgrade to kernel 5.1 or later to address this issue. 
```
(With potential link to github issue if we have one for more details).

Reply:
I mentioned in https://github.com/cilium/cilium/pull/13287#issuecomment-711902104 that the semantics for LRP services should be documented in that svcIP:port frontend tuples are not translated to the backend for the netns where the backend itself is running.

Reply:
I re-worded the note.